lib/AST/Expr.cpp

```diff
@@ -2166,6 +2182,55 @@ void KeyPathExpr::Component::setSubscriptIndexHashableConformances(
   }
 }
 
+void InterpolatedStringLiteralExpr::forEachSegment(llvm::function_ref<void(SegmentInfo)> callback) {
+  auto appendingExpr = getAppendingExpr();
+  if (SemanticExpr) {
+    SemanticExpr->forEachChildExpr([&](Expr *subExpr) -> Expr * {
+      if (auto tap = dyn_cast_or_null<TapExpr>(subExpr)) {
+        appendingExpr = tap;
+        return nullptr;
+      }
+      return subExpr;
+    });
+  }
+  
+  for (auto stmt : appendingExpr->getBody()->getElements()) {
+    if (auto expr = stmt.dyn_cast<Expr*>()) {
+      if (auto apply = dyn_cast<ApplyExpr>(expr)) {
+        SegmentInfo segment;
+        
+        segment.arg = apply->getArg();
+        segment.appendMethod = getCalledDeclRef(apply);
+        
+        DeclName name;
+        if (segment.appendMethod) {
+          name = segment.appendMethod.getDecl()->getFullName();
+        } else if (auto unresolvedDot = dyn_cast<UnresolvedDotExpr>(apply->getFn())) {
+          name = unresolvedDot->getName();
+        }
+        
+        segment.isInterpolation = (name.getBaseName() == "appendInterpolation");
```

Comment:
Can you instead add this identifier and others to Identifiers.def so that it appears as a member of the ASTContext?

Reply:
This identifier actually is in Identifiers.def. I don't use it here because I don't think I can reliably get to an `ASTContext` from this node, and I didn't want to pass a context parameter just to get that identifier. Should I just bite the bullet and do it anyway?